lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -60,6 +60,22 @@ def version
 
       sig { params(changes: T::Array[{ uri: String, type: Integer }]).void }
       def workspace_did_change_watched_files(changes)
+        constants = changes.filter_map do |change|
+          path = change[:uri].gsub("file://", "")
+
+          entries = T.must(@index).entries_for(path, RubyIndexer::Entry::Class, RubyIndexer::Entry::Module)
```

Comment:
Using `grep`, getting subclasses and then filtering them out feels weird to me. Wdyt of an API like this where you have a rest arg and use `==` instead? Or making the second argument optional and returning all entries by default?

Reply:
You can just use the parent class to filter both at the same time.

```suggestion
          entries = T.must(@index).entries_for(path, RubyIndexer::Entry::Namespace)
```

Reply:
I can't only use the parent class because it includes singleton classes. So I'm trying to avoid this filtering with a different API in Ruby LSP https://github.com/Shopify/tapioca/compare/main...andyw8/tapioca-lsp#diff-8243d0258619c7955b4277251f121d4b6788da80bcae39fe5130f27019d143bbR64

Reply:
I think for our use case being able to omit the types and do our own filtering in tapioca makes sense. Will you accept that change in Ruby LSP?

Reply:
I mean, we created that API for the Tapioca add-on, so we can change it to whatever makes sense.